### PR TITLE
Improve attribution and appearance of credits tab

### LINKED ISSUE #2012: Improve appearance and behavior of credits/attribution page on a casebook
Followup to https://github.com/harvard-lil/h2o/issues/1942

* Foreground contributions from earlier authors rather than burying them in the credits paragraph
* Don't include unpublished drafts in attributions list
* Never link to unpublished sections 
* Don't unnecessarily expand TOC subtrees to the point that the list becomes unwieldy

### PR DESCRIPTION
**Pull request by lizadaly**
fixes #2012 

The Credits page on production has the following limitations:

* Authors who contributed to casebooks further back in the ancestry tree get limited credit.
* Casebook TOCs are expanded in full depth, making for a cluttered display that does not add information.
* Oftentimes the immediate parent of a casebook (the one that was cloned from) is a non-public draft, leading to links that 404 or return permission denied.
* The labeling can be confusing (#1814).
* Depending on the user's context, the Credits tab may apply to the whole casebook or just the last-viewed section, but this distinction is opaque.

In addition, the Credits view has some complex business logic but no tests.

This PR:

* More clearly assigns credit to earlier authors.
* Clarifies and links the original document sources from which those contributions were drawn.
* Limits the the information to only the most relevant resource pages.
* Always apply at the Casebook level.
* Better handle cases where previous revisions were drafts that were subsequently published, or were never published drafts.
* Adds a test suite.

I'm sure there are still some edge cases here but I think this smooths over some of the most common flaws in the status quo.

## Before

Casebooks with lots of contributors generated hugely long lists, and obscured individual contributions. Many of these links 404 or lead to login pages because they refer to now-private drafts.

imageimage

## After 

This shows the immediate parent casebook, plus an additional section below for earlier contributors that includes the top-level casebook or sections that were cloned:

image

If we explore the n-1 casebook, the immediate parent, now the two earlier contributors are foregrounded, and the specific resources are listed:

image

### CODE REVIEW

File: web/main/templates/includes/credits.html [hunk 1]
```diff
@@ -1,44 +1,71 @@
 <section class="credits">
     <div>
         {% if contributing_casebooks %}
-        <h3>Content from the following sources has been used in the creation of this {{type}}:</h3>
+        <h3>Content from the following sources has been used in the creation of this casebook</h3>
```
Comment (bensteinberg) [inline web/main/templates/includes/credits.html]: This looks slightly odd to me without a colon or a period at the end.

**Review by bensteinberg — APPROVED** [review]
🔥
